PWGHF/TableProducer/candidateSelectorXicToPKPi.cxx

```diff
@@ -239,15 +290,45 @@ struct HfCandidateSelectorXicToPKPi {
 
       if (pidXicToPKPi == 0 && pidXicToPiKP == 0) {
         hfSelXicToPKPiCandidate(statusXicToPKPi, statusXicToPiKP);
+        if (applyMl) {
+          hfMlXicToPKPiCandidate(outputMlXicToPKPi, outputMlXicToPiKP);
+        }
         continue;
       }
-
       if ((pidXicToPKPi == -1 || pidXicToPKPi == 1) && topolXicToPKPi) {
         statusXicToPKPi = 1; // identified as Xic->pKpi
       }
       if ((pidXicToPiKP == -1 || pidXicToPiKP == 1) && topolXicToPiKP) {
         statusXicToPiKP = 1; // identified as Xic->piKp
       }
+      SETBIT(statusXicToPKPi, aod::SelectionStep::RecoPID);
+      SETBIT(statusXicToPiKP, aod::SelectionStep::RecoPID);
+
+      if (applyMl) {
+        // ML selections
+        bool isSelectedMlXicToPKPi = false;
+        bool isSelectedMlXicToPiKP = false;
+
+        if (statusXicToPKPi > 0) {
+          std::vector<float> inputFeaturesXicToPKPi = hfMlResponse.getInputFeatures(candidate, trackPos1, trackNeg, trackPos2);
+          isSelectedMlXicToPKPi = hfMlResponse.isSelectedMl(inputFeaturesXicToPKPi, ptCand, outputMlXicToPKPi);
+        }
+        if (statusXicToPiKP > 0) {
```

Comment:
Same as above.

Since no variables dependent on the mass hypothesis are used in `HfMlResponseXicToPKPi.h`, you could consider to have only one ML application, common to both cases. You can leave the code as it is if you plan to add those variables at a later stage.